protocol/x/clob/keeper/match_state.go

```diff
@@ -7,12 +7,49 @@ import (
 	"github.com/dydxprotocol/v4-chain/protocol/x/clob/types"
 )
 
-// GetLastTradePriceForPerpetual gets the last trade price for a perpetual.
-func (k Keeper) GetLastTradePriceForPerpetual(
+// GetTradePricesForPerpetual gets the maximum and minimum traded prices for a perpetual for the
+// current block.
+// These prices are intended to be used for improved conditional order triggering in the EndBlocker.
+func (k Keeper) GetTradePricesForPerpetual(
+	ctx sdk.Context,
+	perpetualId uint32,
+) (
+	minTradePriceSubticks types.Subticks,
+	maxTradePriceSubticks types.Subticks,
+	found bool,
+) {
+	minTradePriceSubticks, found = k.getMinTradePriceForPerpetual(ctx, perpetualId)
+	if !found {
+		return 0, 0, false
+	}
+
+	maxTradePriceSubticks, _ = k.getMaxTradePriceForPerpetual(ctx, perpetualId)
+	return minTradePriceSubticks, maxTradePriceSubticks, true
```

Comment:
Don't think this is likely to occur because of the way you're writing to the store for min/max price trades, but the way this function is written we get a weird result if `getMaxTradePriceForPerpetual`'s found value is false, but the min trade price is found.

We would end up returning `minTradePriceSubticks, 0, true`